src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -192,7 +213,7 @@ private async Task<string> SetupInternalAsync(SetupContext context)
                         // unless the recipe is executing?
 
                         _logger.LogError(e, "An error occurred while initializing the datastore.");
-                        context.Errors.Add("DatabaseProvider", S["An error occurred while initializing the datastore: {0}", e.Message]);
+                        context.Errors.Add(String.Empty, S["An error occurred while initializing the datastore: {0}", e.Message]);
```

Comment:
Why String.Empty here?

Reply:
If the tenant database is preconfigured, we never show the connection fields even if the database provider has an error. In this case, we make it empty string and then we render that error to the user so they can see the error. Otherwise, the model state will have an error, but the user will not be able to see it